src/App.tsx

```diff
@@ -15,48 +16,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((item1, item2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return item1.localeCompare(item2);
+
+      case SortType.LENGTH:
+        return item1.length - item2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const goods = getReorderedGoods(goodsFromServer, { sortType, isReversed });
+
+  const handleAlphabetSort = () => {
+    setSortType(SortType.ALPHABET);
+    setIsReversed(false);
+  };
+
+  const handleLengthSort = () => {
+    setSortType(SortType.LENGTH);
+    setIsReversed(false);
+  };
+
+  const handleReverse = () => {
+    setIsReversed(!isReversed);
```

Comment:
It's best practice to rely on actual state when use `setState` with relay on value
```suggestion
    setIsReversed(prev => !prev);
```